freqtrade/rpc/api_server/ws/channel.py

```diff
@@ -170,9 +189,8 @@ async def disconnect_all(self):
         Disconnect all Channels
         """
         with self._lock:
-            for websocket, channel in self.channels.copy().items():
-                if not channel.is_closed():
-                    await channel.close()
+            for websocket in self.channels.copy().keys():
+                await self.on_disconnect(websocket)
 
             self.channels = dict()
```

Comment:
i think this shouldn't be done (setting channels to an empty dict here).

assuming one of the channels issues an exception on close (maybe "closed already" due to some timing issue) - then we'd "lose" all other channels - and never try to disconnect from them again.

obviously our assumption is that the channels dict is empty after this method - but if it's not for whatever reason (maybe a parallel reconnect) - this could become dangerous.